jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsAckConnectorsSpec.scala

```diff
@@ -364,5 +364,85 @@ class JmsAckConnectorsSpec extends JmsSpec {
       }
       resultList.to[SortedSet] should contain theSameElementsAs numsIn.map(_.toString)
     }
+
+    "shutdown when waiting to acknowledge messages" in withServer() { ctx =>
+      val connectionFactory = new ActiveMQConnectionFactory(ctx.url)
+
+      val jmsSink: Sink[JmsTextMessage, Future[Done]] = JmsProducer(
+        JmsProducerSettings(connectionFactory).withQueue("numbers")
+      )
+
+      val (publishKillSwitch, _) = Source
+        .unfold(1)(n => Some(n + 1 -> n))
+        .throttle(15, 1.second, 2, ThrottleMode.shaping) // Higher than consumption rate.
+        .viaMat(KillSwitches.single)(Keep.right)
+        .alsoTo(Flow[Int].map(n => JmsTextMessage(n.toString).withProperty("Number", n)).to(jmsSink))
+        .toMat(Sink.seq)(Keep.both)
+        .run()
+
+      val jmsSource: Source[AckEnvelope, KillSwitch] = JmsConsumer.ackSource(
+        JmsConsumerSettings(connectionFactory).withSessionCount(5).withBufferSize(0).withQueue("numbers")
+      )
+
+      val resultQueue = new LinkedBlockingQueue[String]()
+
+      val (killSwitch, streamDone) = jmsSource
+        .throttle(10, 1.second, 2, ThrottleMode.shaping)
+        .toMat(
+          Sink.foreach { env =>
+            resultQueue.add(env.message.asInstanceOf[TextMessage].getText)
+          //message not acknowledged
+          }
+        )(Keep.both)
+        .run()
+
+      // Need to wait for the stream to have started and running for sometime.
+      Thread.sleep(2000)
+
+      killSwitch.shutdown()
+      publishKillSwitch.shutdown()
+
+      streamDone.futureValue shouldBe Done
```

Comment:
The test was designed so we can validate the result against the generated input. If we don't do the validation we may not need that complicated generation on line 375.

Reply:
Yes I'll simplify 👍